test/distributed/optim/test_apply_optimizer_in_backward.py

```diff
@@ -137,3 +137,19 @@ def test_multiple_optim_for_params(self) -> None:
             [model, model_with_opt_in_bwd],
             [opt_0, opt_1],
         )
+
+    def test_get_optimizers_in_backward(self):
+        # Create a simple test model
+        class TestModel(torch.nn.Module):
+            def __init__(self):
+                super(TestModel, self).__init__()
+                self.linear1 = torch.nn.Linear(10, 5)
+                self.linear2 = torch.nn.Linear(5, 2)
+
+        model = TestModel()
+
+        # Apply optimizers in backward
+        _apply_optimizer_in_backward(torch.optim.SGD, model.parameters(), {'lr': 0.01})
+        result = set(_get_optimizers_in_backward(model))
+        expected = set([optim for p in model.parameters() for optim in p._in_backward_optimizers])
```

Comment:
If the new API returns a list, should we be testing the order too?

Reply:
I don't think the order matters too much - if it does, we should probably be returning something like a map of FQN to the optimizer. @fegin , do you expect the order to matter for checkpointing?

and if order indeed does not matter, is there an alternative return type than list?

Reply:
The order does not matter. If users want to know what is optimized by which optimizer, users can just use `optimzer.param_groups`.